src/datasets/features/features.py

```diff
@@ -265,7 +277,28 @@ def _dtype_error_msg(dtype, pa_dtype, examples=None, urls=None):
     )
 
 
-def _cast_to_python_objects(obj: Any, only_1d_for_numpy: bool, optimize_list_casting: bool) -> tuple[Any, bool]:
+CastableOjbect = Union[
+    np.ndarray,
+    torch.Tensor,
+    tf.Tensor,
+    jnp.ndarray,
+    PIL.Image.Image,
+    pd.Series,
+    pd.DataFrame,
+    pd.Timestamp,
+    pd.Timedelta,
+    Mapping,
+    dict,
+    list,
+    tuple,
+]
```

Comment:
not sure we can use torch, tf, etc. here since they are optional dependencies

Reply:
My bad, we can use forward references for optional runtime dependencies.
```suggestion
CastableOjbect = Union[
    "np.ndarray",
    "torch.Tensor",
    "tf.Tensor",
    "jnp.ndarray",
    "PIL.Image.Image",
    pd.Series,
    pd.DataFrame,
    pd.Timestamp,
    pd.Timedelta,
    Mapping,
    dict,
    list,
    tuple,
]
```